jdaviz/core/marks.py

```diff
@@ -233,23 +347,82 @@ def __init__(self, shadowing, shadow_width=1, **kwargs):
                          colors=[kwargs.pop('color', 'white')],
                          **kwargs)
 
-        # sync initial values
-        for attr in self._sync_traits:
-            setattr(self, attr, getattr(shadowing, attr))
-
-        # keep values synced when traits on shadowing object change
-        shadowing.observe(self._on_shadowing_changed)
+        self._setup_shadowing(shadowing,
+                              ['scales', 'x', 'y', 'visible', 'line_style', 'marker'],
+                              ['stroke_width', 'marker_size'])
 
     def _on_shadowing_changed(self, change):
-        attr = change['name']
-        if attr[0] in ['stroke_width', 'marker_size']:
-            value = change['new'] + self._shadow_width if change['new'] else 0
-        elif attr not in self._sync_traits:
+        super()._on_shadowing_changed(change)
+        if change['name'] in ['stroke_width', 'marker_size']:
+            # apply the same, but increased by the shadow width
+            setattr(self, change['name'],
+                    change['new'] + self._shadow_width if change['new'] else 0)
+        return
+
+
+class ShadowLabelFixedY(Label, ShadowMixin):
+    """
+    Label whose position shadows that of a parent ``shadowing`` line and will flip alignment
+    based on whether it is left or right of the center of the viewer.
+    """
+    def __init__(self, viewer, shadowing, shadow_traits=['visible'],
+                 y=0.95, point_index=0, **kwargs):
+        super().__init__(**kwargs)
+        self._viewer = viewer
+        self.y = [y]
+        self.scales['y'] = LinearScale(min=0, max=1)
+        self._point_index = point_index
+        self._setup_shadowing(shadowing,
+                              shadow_traits,
+                              ['x', 'scales', 'labels', 'colors'])
+
+        viewer.state.add_callback("x_min", lambda x_min: self._update_align())
+        viewer.state.add_callback("x_max", lambda x_max: self._update_align())
+
+    def _force_redraw(self):
+        # TODO: bug in bqplot that change in align/colors traitlet doesn't update immediately,
```

Comment:
Does that mean it *eventually* updates but just not fast enough for this feature? Maybe there is a way to "flush the buffer"?

Reply:
no, by immediately I don't mean time, but rather that it doesn't update until the next interaction would force it to update anyways (at which point it does pick up on the changes)